core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -176,7 +178,8 @@ public boolean exists(AlluxioURI path, ExistsOptions options)
     FileSystemMasterClient masterClient = mFileSystemContext.acquireMasterClient();
     try {
       // TODO(calvin): Make this more efficient
```

Comment:
The issue was with using exceptions for control flow caused by reusing the `getStatus` method for exists.